### ISSUE TITLE
Relation between languages and WALS genera is gone

### ISSUE THREAD

**Issue by xrotwang** [phoible/dev#145]
Just to confirm: The relation between languages present in PHOIBLE and the WALS genus they fall in is no longer maintained as part of the data, right?

I think it makes sense to drop this relation, although the concept of WALS genera may be a good thing to have, e.g. for sampling strategies. Anyway, the best way to re-introduce this would presumably be through a corresponding notion of genus in Glottolog, possibly seeded from WALS. Will talk to Harald about this.

**Comment by xrotwang** [comment]
@d97hah what do you think about WALS-like genera in Glottolog?

**Comment by d97hah** [comment]
It's not a well-defined notion so it can't be implemented. What one could
do as a substitute is to have a lex-stat number attached to each internal
node, computed from e.g. ASJP but ASJP doesn't cover all lgs (yet).

2016-08-25 12:02 GMT+02:00 Robert Forkel <notifications@github.com>:

**Comment by drammock** [comment]
The current PHOIBLE heirarchy of identifiers goes (from most granular to least granular):

> PHOIBLE inventoryID (effectively a doculect ID), glottocode (partial coverage), ISO code (full coverage), WALS genus (? coverage), ethnologue family/stock

I don't object to dropping the WALS genera data from PHOIBLE as long as there is some level of identifier that can serve as a link between PHOIBLE and WALS, so users could merge the WALS genera into a PHOIBLE dataset if they wanted to.  At present the ISO code is sufficient for this task.

**Comment by xrotwang** [comment]
Both, ISO code and Glottocode can be used to merge the WALS genera into a PHOIBLE dataset.

Btw. as far as I can tell, WALS genera **and** ethnologue family/stock have already been dropped from the PHOIBLE data (at least from `data/phoible-by-phoneme.tsv`).

**Comment by bambooforest** [comment]
Indeed we dropped all non-phoible data except identifiers. One issue regarding WALS genera is that I had to hand curate all those data points for which there are languages that are not in phoible. I don't suppose that WALS genera exist for all languages in Glottolog -- so I could add this hand-curated data file somewhere, if we want to keep the link to WALS genera and the inferred points.

**Comment by xrotwang** [comment]
@bambooforest I you were to hand-curate such a file, I think the even greater service would be one that links all PHOIBLE languages to Glottolog sub-groups corresponding to what the WALS genus would have been. This way, you would
- make sure you can find a genus-like sub-group for each language (possibly PHOIBLE does now have languages where the whole family is absent from WALS)
- add information about this (ill-defined according to @d97hah) intermediate grouping of languages.

**Comment by bambooforest** [comment]
Ok. An fyi -- I hand curated the WALS genera back in 2010, when writing the thesis. I haven't done any additions since (so the newer languages won't be mapped to WALS codes or genera).

I agree a more stable approach is warranted if want to have Glottolog sub-groups that correspond to WALS genera. Can we automatically map our existing data to get started? We still have phobile IDs to WALS genera aound and I think for at least half of the current inventories I may even have ISO 639-3 code to WALS code (if they haven't been updated).

**Comment by xrotwang** [comment]
@bambooforest I think I could seed the mapping automatically. Could do this once you completed the mapping to Glottolog and ISO.

**Comment by d97hah** [comment]
My hunch is the following automated strategy might work to a very high
degree (to map an iso-code to a WALS genus):
1. If the iso-code is in WALS, take the genus from there
2. If the iso-code is not in WALS, look at the successively closest
relative of the iso code which is in WALS and assign that genus
3. Leave no genus assigned to the iso-code if it has no relative in WALS

2016-08-29 13:41 GMT+02:00 Steven Moran <notifications@github.com>:

**Comment by xrotwang** [comment]
If the genus concept is so ill-defined that no complete mapping of PHOIBLE languages to such entities in Glottolog can be constructed, I think it would be best to drop this information from PHOIBLE for good. Because otherwise, any analyses using the genus mapping would not only rely on this ill-defined concept but also introduce sampling bias from WALS.

**Comment by drammock** [comment]
I agree about WALS genera potentially *introducing* a sampling bias despite its intent to be a tool to facilitate equal-time-depth sampling.  So although I want end users to be able to integrate WALS genera into a PHOIBLE dataset if they want to, I would not push @bambooforest (or anyone else) to spend a lot of time on expanding/improving what already exists in WALS.

**Comment by bambooforest** [comment]
tl;dr we decided to drop language genealogy from phoible (e.g. WALS genus) and other information like language coordinates (lat and long) because we can get this data (phylogenetic trees, coordinates) from Glottolog via Glottolog codes in our mappings file

https://github.com/phoible/dev/blob/master/mappings/InventoryID-LanguageCodes.tsv
